Decks/Source/LimitedDeck.cpp

```diff
@@ -0,0 +1,40 @@
+#include "LimitedDeck.hpp"
+
+HandDeck::HandDeck(uint maxSize)
+  :_maxSize(maxSize)
+{
+
+}
+
+TableDeck::TableDeck(uint maxSize)
+  :_maxSize(maxSize)
+{
+
+}
+
+void HandDeck::drawCard(Deck &deck)
+{
+  if(cards->size() >= _maxSize)
+  {
+    std::cout << "Vaše ruka je plná" << std::endl;
+  }
+  else
+  {
+    cards->push_back(deck.cards->at(0));
+    deck.removeFirstCard();
+  }
+}
+
+void HandDeck::playCard(std::shared_ptr<TableDeck> tableDeckPointer, Card CardToThrow)
+{
+  if(tableDeckPointer->cards->size() >= tableDeckPointer->_maxSize)
+  {
+    std::cout << "Váš stůl je již plný" << std::endl;
+  }
+  else
+  {
+    uint cardIndex = findCard(CardToThrow);
+    tableDeckPointer->cards->push_back(this->cards->at(cardIndex));

```

Comment:
nie widzę tego jak to będzie zagrywane jak kartę będzie czar, ale bawcie się dalej, zobaczymy co Wam z tego wyjdzie.

Reply:
To jest do zmiany, będzie tu warunek który jeśli sprawdzi że karta jest Monsterem to ją zpushbackuje na stoł a usunie z ręki a jeśli Spellem usunie z ręki i pewno wysle wskaźnik do Walki, z którego Walka od razu skorzysta.

Reply:
Ale to bym potrzebował aktualnych Cards na masterze

Reply:
W zasadach SOLID (czyli jak programować obiektowo) jest to, że nie musimy znać jaka to klasa, jeśli mamy wspólny interfejs od "ojca" to powinno się to dać wykonać na każdej. Ale tak jak mówiłem - piszcie i zobaczymy później co Wam z tego wyjdzie.